arrow-avro/src/codec.rs

```diff
@@ -1857,14 +1839,14 @@ mod tests {
             .expect("promotion should resolve")
     }
 
-    fn mk_primitive(pt: PrimitiveType) -> Schema<'static> {
+    fn mk_primitive<'a>(pt: PrimitiveType) -> Schema<'a> {
         Schema::TypeName(TypeName::Primitive(pt))
```

Comment:
Interesting... `'a` doesn't "attach" to any input arg? Not quite sure how that works in rust...

Reply:
Does this work, out of curiosity?
```suggestion
    fn mk_primitive(pt: PrimitiveType) -> Schema<'_> {
        Schema::TypeName(TypeName::Primitive(pt))
```

Reply:
Nope... might be a case where `'static` actually makes sense?